utils/walk.test.js

```diff
@@ -0,0 +1,68 @@
+const assert = require('assert');
+
+const bcd = require('./bcd');
+const { walk } = require('./index');
+const { lowLevelWalk } = require('./walk');
+
+describe('lowLevelWalk()', function () {
+  it('visits every top-level tree', function () {
+    const expectedPaths = [
+      'api',
+      'browsers',
+      'css',
+      'html',
+      'http',
+      'javascript',
+      'mathml',
+      'svg',
+      'webdriver',
+      'webextensions',
+      'xpath',
+      'xslt',
+    ];
+
+    const steps = Array.from(lowLevelWalk(undefined, undefined, 1));
+    const paths = steps.map(step => step.path);
+    assert.strictEqual(steps.length, expectedPaths.length);
+    assert.deepStrictEqual(paths, expectedPaths);
+  });
+  it('visits every point in the tree', function () {
+    const paths = Array.from(lowLevelWalk()).map(step => step.path);
+    assert.ok(paths.length > 13000);
+  });
+});
+
+describe('walk()', function () {
+  it('should visit deeply nested features', function () {
+    let results = Array.from(walk('html')).map(feature => feature.path);
+    assert.ok(results.includes('html.elements.a.href.href_top'));
+  });
+
+  it('should walk a single tree', function () {
+    let results = Array.from(walk('api.Notification'));
+    assert.strictEqual(results.length, 27);
+    assert.strictEqual(results[0].path, 'api.Notification');
+    assert.strictEqual(results[1].path, 'api.Notification.Notification');
+  });
+
+  it('should walk multiple trees', function () {
+    let results = Array.from(
+      walk(['api.Notification', 'css.properties.color']),
+    );
+    assert.strictEqual(results.length, 28);
+    assert.strictEqual(results[0].path, 'api.Notification');
+    assert.strictEqual(
+      results[results.length - 1].path,
+      'css.properties.color',
+    );
+  });
+
+  it('should yield every feature by default', function () {
```

Comment:
This test (and the similar test of `visit`) are rather slow (>300ms and >50ms on my machine), which Mocha highlights. I think it's mostly the overhead of Mocha, but we could restrict this to some subtree if we want to speed things up.

Reply:
We could also have a unchanging set of BCD-like JSON files for testing only, which could exercise things that are uncommon in BCD or likely to change over time. That would probably be blazing fast to test with.

One might still have one or two smoke tests involving the real data.

Reply:
Testing with a unchanging set will be a lot easier now that we've got the consistent `data` parameter.

Is this something I should come up with as part of this PR?